ci_scripts/jira_scripts/check_jira_status.py

```diff
@@ -0,0 +1,145 @@
+                retries += 1
+                print(
+                    "Failed to get issue due to inactivity, retrying (%d/%d)",
+                    retries,
+                    max_retry,
+                )
+                if retries < max_retry:
+                    jira_connection = get_jira_connection()  # Attempt reconnection
+                else:
+                    raise  # Re-raise the error after exceeding retries
+            else:
+                raise
+
+
+def get_jira_fix_version(jira_metadata):
+    fix_version = (
+        re.search(r"([\d.]+)", jira_metadata.fixVersions[0].name)
+        if jira_metadata.fixVersions
+        else None
+    )
+    return fix_version.group(1) if fix_version else "vfuture"
+
+
+def get_all_jiras_from_file(file_content):
+    """
+    Try to find all jira tickets in the file.
+    Looking for the following patterns:
+    - jira_id=<id>>  # call in is_jira_open
+    - jira_id = <id>  # when jira is constant
+    - https://issues.redhat.com/browse/<id>  # when jira is in a link in comments
+    - pytest.mark.jira(id)  # when jira is in a marker
+
+    Args:
+        file_content (str): The content of the file.
+
+    Returns:
+        list: A list of jira tickets.
+    """
+    issue_pattern = r"([A-Z]+-[0-9]+)"
+    _pytest_jira_marker_bugs = re.findall(
+        rf"pytest.mark.jira.*?{issue_pattern}.*", file_content, re.DOTALL
+    )
+    _is_jira_open = re.findall(rf"jira_id\s*=[\s*\"\']*{issue_pattern}.*", file_content)
+    _jira_url_jiras = re.findall(
+        rf"https://issues.redhat.com/browse/{issue_pattern}.*",
+        file_content,
+    )
+    return set(_pytest_jira_marker_bugs + _is_jira_open + _jira_url_jiras)
+
+
+def get_jiras_from_python_files():
+    jira_found = {}
+    for filename in all_python_files():
+        filename_for_key = re.findall(r"openshift-virtualization-tests/.*", filename)[0]
+        with open(filename) as fd:
+            if unique_jiras := get_all_jiras_from_file(file_content=fd.read()):
+                jira_found[filename_for_key] = unique_jiras
+    return jira_found
+
+
+def main():
+    closed_statuses = get_connection_params().get("resolved_statuses")
+    closed_jiras = {}
+    mismatch_bugs_version = {}
+    jira_ids_with_errors = {}
+    jira_ids_dict = get_jiras_from_python_files()
+    jira_connection = get_jira_connection()
+    for filename in jira_ids_dict:
+        for jira_id in jira_ids_dict[filename]:
+            try:
+                jira_metadata = get_jira_metadata(
+                    jira_id=jira_id, jira_connection=jira_connection
+                )
+                current_jira_status = jira_metadata.status.name.lower()
+                if current_jira_status in closed_statuses:
+                    closed_jiras.setdefault(filename, []).append(
+                        f"{jira_id} [{current_jira_status}]"
+                    )
+                jira_target_release_version = get_jira_fix_version(
```

Comment:
Should we check the fix version for bugs only? Sometimes we use jira ids of Epics / Stories / Tasks - we do not necessarily own them, so we can't make the clones targeter to other versions.
And, for example, if there's a task to fix some automation - to me it looks enough to state in the acceptance criteria that it should be backported, and there's no need to clone the Task just to give it a matching fix version.

Reply:
No. We should not put any tasks/stories/epics to our test automation, specially in non-relevant branches. This should be tracked in jira backlogs only.